src/main/java/cs361/battleships/models/Weapon.java

```diff
@@ -0,0 +1,29 @@
+package cs361.battleships.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.Sets;
+import com.mchange.v1.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+public class Weapon {
```

Comment:
nice, add a weapon class increases the modularity of our code base for future projects!